### PR TITLE
Prevent non-docs stages when a PR only modifies docs/

### PR DESCRIPTION
**Pull request by mikepapadim**
This is an intermediate fix for #8673.
It modifies `Jenkinsfile` to prevent non-doc related stages from running when only modifying in `docs/`.
The idea is to save some CI cycles and speed up the process of augmenting docs.

@areusch @comaniac @tqchen

### CODE REVIEW

**Comment by Hzfengsy** [comment]
Thanks, @mikepapadim. I agree that it is important to save the CI time.

However, tutorials may be changed even though the codes in `docs/` are not changed. For example, when we change the printer(files in `src/` and `include`) may still change the document. These changes are not often happened but they do exist.

**Comment by areusch** [comment]
@mikepapadim this looks much closer to what we want but I think if I'm reading this right, I see:
- whenever a change only modifies `docs/`, we skip all build, unit test, integration test steps

as @Hzfengsy said, we shouldn't ever skip these nodes:
- Sanity Check (this should already be correct in the patch now)
- BUILD: GPU
- docs: GPU

i think we need to find a way to propagate these conditions down to the node level in each stage (e.g. skip each node not listed above when `changeset "docs/**"` doesn't match). i'm not entirely sure how to do this.

**Comment by mikepapadim** [comment]
> @mikepapadim this looks much closer to what we want but I think if I'm reading this right, I see:
> 
> * whenever a change only modifies `docs/`, we skip all build, unit test, integration test steps
> 
> as @Hzfengsy said, we shouldn't ever skip these nodes:
> 
> * Sanity Check (this should already be correct in the patch now)
> * BUILD: GPU
> * docs: GPU
> 
> i think we need to find a way to propagate these conditions down to the node level in each stage (e.g. skip each node not listed above when `changeset "docs/**"` doesn't match). i'm not entirely sure how to do this.

Yes, right now the `changeset` condition is in the stage scope, I will have another look on how to make use of it in `node` scope.

File: docker/install/ubuntu_install_vitis_ai_packages_ci.sh [hunk 1]
```diff
@@ -6,9 +6,9 @@
 # to you under the Apache License, Version 2.0 (the
 # "License"); you may not use this file except in compliance
 # with the License.  You may obtain a copy of the License at
-# 
+#
```
Comment (leandron) [inline docker/install/ubuntu_install_vitis_ai_packages_ci.sh:9]: This seems like an unrelated change.

File: .github/ISSUE_TEMPLATE/documentation.md [hunk 2]
```diff
@@ -19,4 +19,3 @@ Include the title of the document (e.g. "Getting Started with TVM"), and the typ
 If an RFC/discuss post exists, link it here.
 
 Otherwise, specify what actions should be taken to provide additional clarity/readability/reproducibility to the document. Include code snippets from the previous documentation if applicable.
```
Comment (leandron) [inline .github/ISSUE_TEMPLATE/documentation.md:21]: This seems like an unrelated change.
